### PR TITLE
[Enhancement] Automatically populate column statistic cache after Analyze table

### LINKED ISSUE #10141: Could populate column statistic cache after Analyze table manually
## Enhancement
Currently, after  analyze table manually, we only collect column statistic in BE, don't populate column statistic cache in FE until user first query the column. So, for the first query about the column, the query plan maybe bad.
We could add a session variable and populate column statistic cache in FE immediately after Analyze table manually.

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #10141 

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 27 (22.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|8|00.00%|[215, 217, 218, 219, 222, 223, 224, 225]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|4|00.00%|[824, 825, 835, 836]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|2|00.00%|[802, 815]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeManager.java|0|3|00.00%|[152, 185, 186]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|2|4|50.00%|[375, 386]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|3|5|60.00%|[138, 154]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticStorage.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 1]
```diff
@@ -819,13 +819,21 @@ private void handleAnalyzeStmt() throws IOException {
                     new HistogramStatisticsCollectJob(db, table, analyzeStmt.getColumnNames(),
                             StatsConstants.AnalyzeType.HISTOGRAM, StatsConstants.ScheduleType.ONCE,
                             analyzeStmt.getProperties()));
+
+            //Could populate column statistic cache after Analyze table manually
+            GlobalStateMgr.getCurrentStatisticStorage().expireHistogramStatistics(table.getId(), analyzeStatus.getColumns());
+            GlobalStateMgr.getCurrentStatisticStorage().getHistogramStatistics(table, analyzeStatus.getColumns());
```
Comment (liuyehcf) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:825]: Is it means re-creating after expired? Maybe better to put them inside of refreshXXX like `GlobalStateMgr.getCurrentAnalyzeMgr().refreshBasicStatisticsCache`

**Review by liuyehcf — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 2]
```diff
@@ -819,13 +819,21 @@ private void handleAnalyzeStmt() throws IOException {
...
             analyzeStatus = statisticExecutor.collectStatistics(
                     StatisticsCollectJobFactory.buildStatisticsCollectJob(db, table, null,
                             analyzeStmt.getColumnNames(),
                             analyzeStmt.isSample() ? StatsConstants.AnalyzeType.SAMPLE :
                                     StatsConstants.AnalyzeType.FULL,
                             StatsConstants.ScheduleType.ONCE, analyzeStmt.getProperties()));
+
+            //Could populate column statistic cache after Analyze table manually
+            GlobalStateMgr.getCurrentStatisticStorage().expireColumnStatistics(table, analyzeStatus.getColumns());
+            GlobalStateMgr.getCurrentStatisticStorage().getColumnStatistics(table, analyzeStatus.getColumns());
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:836]: This is a async load method, we need a sync method.

**Review by Seaven — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 13 / 76 (17.11%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|11|00.00%|[215, 217, 218, 219, 220, 221, 224, 225, 226, 227, 228]|
|:large_blue_circle:|com/starrocks/persist/EditLog.java|0|5|00.00%|[803, 804, 817, 818, 819]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeManager.java|0|12|00.00%|[143, 147, 151, 152, 153, 155, 179, 183, 188, 189, 190, 192]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/ColumnBasicStatsCacheLoader.java|0|8|00.00%|[70, 73, 74, 75, 77, 78, 79, 80]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|8|30|26.67%|[117, 120, 121, 124, 125, 128, 129, 130, 131, 132, 134, 135, 137, 138, 139, 140, 141, 143, 145, 146, 175, 182]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticStorage.java|1|3|33.33%|[17, 25]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|5|40.00%|[822, 823, 825]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|2|2|100.00%|[]|

**Review by liuyehcf — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by kangkaisen — APPROVED** [review]

**Comment by kangkaisen** [comment]
Will also fix https://github.com/StarRocks/starrocks/issues/10157

**Pull request merged**